lib/models/LifecycleConfiguration.js

```diff
@@ -412,6 +412,252 @@ class LifecycleConfiguration {
+        const usedStorageClasses = [];
+        for (let i = 0; i < rule.NoncurrentVersionTransition.length; i++) {
+            const t = rule.NoncurrentVersionTransition[i]; // Transition object
+            const noncurrentDays = t.NoncurrentDays && t.NoncurrentDays[0];
+            const storageClass = t.StorageClass && t.StorageClass[0];
+            if (noncurrentDays === undefined || storageClass === undefined) {
+                return Object.assign(result, { error: errors.MalformedXML });
+            }
+            let error = this._checkDays({
+                days: noncurrentDays,
+                field: 'NoncurrentDays',
+                ancestor: 'NoncurrentVersionTransition',
+            });
+            if (error) {
+                return Object.assign(result, { error });
+            }
+            error = this._checkStorageClasses({
+                storageClass,
+                usedStorageClasses,
+                ancestor: 'NoncurrentVersionTransition',
+                prefix: this._getRulePrefix(rule),
+            });
+            if (error) {
+                return Object.assign(result, { error });
+            }
+            usedStorageClasses.push(storageClass);
+            result.nonCurrentVersionTransition.push({
+                noncurrentDays: Number.parseInt(noncurrentDays, 10),
+                storageClass,
+            });
+        }
+        return result;
+    }
+
+    /**
+     * Parses the Transition value
+     * @param {object} rule - Rule object from Rule array from this._parsedXml
+     * @return {object} - Contains error if parsing failed, otherwise contains
+     * the parsed transition array
+     *
+     * Format of result:
+     * result = {
+     *      error: <error>,
+     *      transition: [
+     *          {
+     *              days: <days>,
+     *              date: <date>,
+     *              storageClass: <storage-class>,
+     *          },
+     *          ...
+     *      ]
+     * }
+     */
+    _parseTransition(rule) {
+        const result = { transition: [] };
+        const usedStorageClasses = [];
+        let usedTimeType = null;
+        for (let i = 0; i < rule.Transition.length; i++) {
+            const t = rule.Transition[i]; // Transition object
+            const days = t.Days && t.Days[0];
+            const date = t.Date && t.Date[0];
+            const storageClass = t.StorageClass && t.StorageClass[0];
+            if (days === undefined && date === undefined) {
+                return Object.assign(result, { error: errors.MalformedXML });
+            }
+            if (days !== undefined && date !== undefined) {
+                return Object.assign(result, { error: errors.MalformedXML });
+            }
+            if (storageClass === undefined) {
+                return Object.assign(result, { error: errors.MalformedXML });
+            }
+            let error = null;
+            if (days !== undefined) {
+                error = this._checkTimeType({
```

Comment:
I don't understand the need for this function, since above we check `if (days !== undefined && date !== undefined)`, which looks good enough to me, except we might want to add a specific error message, if I read this correctly?